web-admin/src/features/projects/status/waitForDeployment.ts

```diff
@@ -0,0 +1,221 @@
+} from "@rilldata/web-admin/client";
+import type { CompoundQueryResult } from "@rilldata/web-common/features/compound-query-result";
+import {
+  fetchResources,
+  ResourceKind,
+  SingletonProjectParserName,
+  useResourceV2,
+} from "@rilldata/web-common/features/entity-management/resource-selectors";
+import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+import {
+  V1ReconcileStatus,
+  type V1Resource,
+} from "@rilldata/web-common/runtime-client";
+import { derived, type Unsubscriber } from "svelte/store";
+
+const PollTime = 1000;
+
+export class WaitForDeployment {
+  private errored = false;
+  private redirectToResource: V1Resource | undefined;
+
+  // The user is 1st redirected to `invite` page. They might take some time on there or be very quick.
+  // We use this boolean to identify if a deployment already succeeded by the time user lands on dashboards page.
+  private shouldRedirect = false;
+
+  private readonly unsub: Unsubscriber;
+
+  private static instance: WaitForDeployment | undefined;
+
+  private constructor(
+    private readonly organization: string,
+    private readonly project: string,
+  ) {
+    this.unsub = deploymentListener(organization, project).subscribe(
+      async (status) => {
+        if (status.isFetching) return;
+        this.unsub?.();
+        if (status.error) {
+          this.errored = true;
+        } else if (status.data) {
+          const resources = await fetchResources(queryClient, status.data);
+          // prefer a metrics view, if there are none select a custom dashboard
+          this.redirectToResource =
+            resources.find(
+              (r) => r.meta?.name?.kind === ResourceKind.MetricsView,
+            ) ??
+            resources.find(
+              (r) => r.meta?.name?.kind === ResourceKind.Dashboard,
+            );
+        }
+
+        if (this.shouldRedirect) {
+          // if the user is on dashboards page before deployment was completed.
+          this.handlePostDeployment();
+        } else {
+          // else do not do anything until user is out of invite page.
+          this.shouldRedirect = true;
+        }
+      },
+    );
+  }
+
+  public static create(organization: string, project: string) {
+    this.instance = new WaitForDeployment(organization, project);
+  }
+
+  public static wait() {
+    if (!this.instance) return;
+
+    if (!this.instance.shouldRedirect) {
+      // user landed on dashboards page before deployment is completed.
+      this.instance.shouldRedirect = true;
+    } else {
+      // deployment is already complete.
+      this.instance.handlePostDeployment();
+    }
+
+    this.instance = undefined;
+  }
+
+  private handlePostDeployment() {
+    if (this.errored) {
+      eventBus.emit("notification", {
+        message: "Failed to deploy project",
+      });
+      void goto(`/${this.organization}/${this.project}`);
```

Comment:
So if the deployment errored, we go to the project homepage, right? You say the status page in the PR description. But anyways both pages in their current form show a 500 error:

<img width="1474" alt="image" src="https://github.com/user-attachments/assets/f840510f-8168-49e4-b952-6d3b2a70ca52">

Reply:
Ah yes. It should redirect to status page.